ompi/attribute/attribute_predefined.c

```diff
@@ -194,11 +191,12 @@ static int create_comm(int target_keyval, bool want_inherit)
     ompi_attribute_fn_ptr_union_t del;
 
     keyval = -1;
-    copy.attr_communicator_copy_fn = (MPI_Comm_internal_copy_attr_function*)
-        (want_inherit ? MPI_COMM_DUP_FN : MPI_COMM_NULL_COPY_FN);
+    copy.attr_communicator_copy_fn = (MPI_Comm_internal_copy_attr_function
```

Comment:
What happened here?

Reply:
This we can fix by using proper function pointer typedefs. I have a PR up to address it.